src/VirtoCommerce.PricingModule.Web/Scripts/blades/assignment-detail.js

```diff
@@ -141,9 +146,80 @@ angular.module('virtoCommerce.pricingModule')
     };
 
     // actions on load
-    $scope.catalogs = catalogs.query();
-    pricelists.search({ take: 1000 }, function (result) {
-    	$scope.pricelists = result.results;
-    });
+    $scope.fetchCatalogs = ($select) => {
+        while (blade.isLoading) sleep(100);
+        if ($scope.catalogs.length == 0) {
+            $select.page = 0;
+
+            if (blade.currentEntity.catalogId) {
+                let criteria = {
+                    CatalogIds: [blade.currentEntity.catalogId]
+                }
+                catalogs.search(criteria, (data) => {
+                    $scope.catalogs = data.results;
+                    $scope.fetchNextCatalogs($select);
+                });
+            }
+            else {
+                $scope.fetchNextCatalogs($select);
+            }
+        }
+    }
+    $scope.fetchNextCatalogs = ($select) => {
+        let criteria = {
+            SearchPhrase: $select.search,
+            take: $scope.pageSize,
+            skip: $select.page * $scope.pageSize
+        }
+
+        catalogs.search(criteria, (data) => {
+            $scope.catalogs = $scope.catalogs.concat(data.results);
+            $select.page++;
+
+            if ($scope.catalogs.length < data.totalCount) {
+                $scope.$broadcast('scrollCompleted');
+            }
+        });
+    }
+
+
+    $scope.fetchPricelists = ($select) => {
+        while (blade.isLoading) sleep(100);
```

Comment:
can call fetch from initializeBlade instead of this sleep?

Reply:
Need good advice, how to obtain $select during initializing, because we need to store the last fetched page.

Reply:
I think this is the best way to wait while EntityId will be filled in initialization function. The $scope.fetchPricelist function called from angular template that rendered before currentEntity.pricelistId filled.

Reply:
Another way is to set CatalogId before Initialisation